daffodil-udf/src/test/scala/org/apache/daffodil/udf/TestUdfs.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.daffodil.udf
+
+import org.junit.AfterClass
+import org.junit.Test
+import junit.framework.Assert._
+import java.net.URLClassLoader
+import org.apache.daffodil.util.Misc
+import org.apache.daffodil.udf
+import java.net.URL
+import collection.JavaConverters._
+
+class TestUdfs {
+  @Test def test_noUdfsLoaded() {
+    try {
+      lazy val addURL = classOf[URLClassLoader].getDeclaredMethod("addURL", classOf[URL]);
+      print(Misc.classPath)
+      val ucl: URLClassLoader = new URLClassLoader(Array(new URL("file:/to/nowhere/")))
+      /*
+            addURL.setAccessible(true)
+      addURL.invoke(ucl, url)
+      val urlsOfUcl = ucl.getURLs
+      val uclAsCL = ucl.asInstanceOf[ClassLoader]
+      val rsrcOfUCL = uclAsCL.getResources("META-INF/services/org.apache.daffodil.udf.UDFunctionProvider")*/
+      val uclAsCL = ucl.asInstanceOf[ClassLoader]
+      val emptyCL = ucl.getParent.getParent
+      val udfs = new UDFunctionService(emptyCL)
+      val errList = udfs.getErrors
+      val warningsList = udfs.getWarnings
+      assertEquals(1, errList.size())
+      assertEquals(0, warningsList.size())
+      assertTrue(errList.get(0).contains("No user defined functions found."))
+    } catch {
+      case e: Exception => e.printStackTrace()
```

Comment:
Remove try/catch. Just let the test fail if a throw occurs by letting it propagate.